app/models/vm_scan.rb

```diff
@@ -466,7 +466,8 @@ def process_cancel(*args)
     _log.info("job canceling, #{options[:message]}")
 
     begin
-      delete_snapshot(context[:snapshot_mor])
+      vm = VmOrTemplate.find_by(:id => target_id)
+      process_delete_evm_snapshot(vm)
```

Comment:
The replaced ```delete_snapshot``` method has some logics about user_event and cleaning up snapshot descriptions. They are all missed in your new defined ```process_delete_evm_snapshot``` method. You sure they don't need?

Reply:
On further review you are correct @hsong-rh.  Reworking this PR now and will re-test.